src/node/interfaces.cpp

```diff
@@ -854,6 +856,13 @@ class MinerImpl : public Miner
         return TestBlockValidity(state, chainman().GetParams(), chainman().ActiveChainstate(), block, chainman().ActiveChain().Tip(), /*fCheckPOW=*/false, check_merkle_root);
     }
 
+    std::unique_ptr<CBlockTemplate> createNewBlock(const CScript& scriptPubKeyIn) override
+    {
+        LOCK(::cs_main);
+        // TODO pass mempool sanely
+        return BlockAssembler{chainman().ActiveChainstate(), &*(context()->mempool)}.CreateNewBlock(scriptPubKeyIn);
```

Comment:
f04cf9f75054573c204245f7bb1e6813cce1fed8: I guess I should change the `BlockAssembler` constructor

Reply:
re: https://github.com/bitcoin/bitcoin/pull/30200#discussion_r1621067334

> I guess I should change the `BlockAssembler` constructor

In commit "rpc: call CreateNewBlock via miner interface" (f04cf9f75054573c204245f7bb1e6813cce1fed8)

I don't actually understand the TODO here. The way the mempool is passed seems ok. You could do mempool.get() instead of &*mempool, but both should be equivalent.

Reply:
Thanks, replaced it with `get()`.